controllers/controllervm_controller.go

```diff
@@ -152,34 +157,33 @@ func (r *ControllerVMReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
 		// TODO: for now single network and hardcode controller ips to 192.168.25.1X
 
 		hostname := fmt.Sprintf("controller-%d", i)
-		ip := fmt.Sprintf("192.168.25.1%d", i)
 		networkDataSecretName := fmt.Sprintf("%s-%s-networkdata", instance.Name, hostname)
-		templateParameters["ControllerIP"] = ip
-
-		controllerDetails = append(controllerDetails,
-			ospdirectorv1beta1.Host{
-				Hostname:          hostname,
-				DomainName:        hostname,
-				DomainNameUniq:    fmt.Sprintf("%s-%s", hostname, instance.UID[0:4]),
-				IPAdress:          ip,
-				NetworkDataSecret: networkDataSecretName,
-			},
-		)
 
-		sts = append(sts, common.Template{
-			Name:           fmt.Sprintf("%s-%s-networkdata", instance.Name, hostname),
-			Namespace:      instance.Namespace,
-			Type:           common.TemplateTypeNone,
-			InstanceType:   instance.Kind,
-			AdditionalData: map[string]string{"networkdata": "/controllervm/cloudinit/networkdata"},
-			Labels:         secretLabels,
-			ConfigOptions:  templateParameters,
-		})
-	}
+		controllerDetails[hostname] = ospdirectorv1beta1.Host{
+			Hostname:          hostname,
+			DomainName:        hostname,
+			DomainNameUniq:    fmt.Sprintf("%s-%s", hostname, instance.UID[0:4]),
+			IPAdress:          fmt.Sprintf("192.168.25.1%d", i),
+			NetworkDataSecret: networkDataSecretName,
+		}
 
-	err = common.EnsureSecrets(r, instance, sts, &envVars)
-	if err != nil {
-		return ctrl.Result{}, nil
+		templateParameters["ControllerIP"] = controllerDetails[hostname].IPAdress
```

Comment:
I had noticed this last week. No need to fix it here but I think we should rename this IPAddress.

Reply:
yes sounds good. we can do it with the IPAM integration